### DISCUSSION TITLE
Combine features of `blog` and `docs` pages?

### DISCUSSION THREAD

**Discussion by ghost** [facebook/docusaurus#5937] [General] [answered]
Is there an easy way to combine features of `blog` and `docs` pages? 

I'm building a personal portfolio style site, and what I really need is a combo of the two. For example, it'd be great to have from `blog` defaults things like:
- date of publishing
- tags
- index of recent posts
- index of tags
_and_ from `docs` defaults things like:
- sidebar
- table of contents
- versioning

I don't specifically need all of this, but I think it'd a powerful in general to be able to combine all/most features at will. I see that [`docs` do have tags](https://docusaurus.io/docs/create-doc#doc-tags), but I can't find a way to easily cross-use other features.

Anyone know a way? 

---

**A brief follow up:** I noticed [in the `plugin-cotent-blog` docs](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-blog#hide_table_of_contents) there's a `hide_table_of_contents` with `false` as the default. But this doesn't seem to be implemented, since a blog post without that set to `true` (e.g. in the starter project) doesn't have a table of contents. Strange.

**Comment by ghost** [comment]
Partly solved, for the case of table of contents in a blog post. None of the examples in the started project have any headers, so no table of contents appears! When you add headers, it shows up :)

**Comment by slorber** [comment, marked as answer]
Plugins are independant and don't see each others code, so one feature of a plugin is not usable in another.

If you need sidebar and versioning, you probably want docs.

Displaying a date of publishing can be added by using a custom frontmatter and swizzle

We could add a docs flat list view page and you'd be able to preset it as "recent docs" if you really want to.

In any case, what you are looking for is something custom. We do blog, we do docs, not a custom mix in between. Be prepared to write some code to handle your specific use-case
